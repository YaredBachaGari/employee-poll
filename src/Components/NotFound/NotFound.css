.notfound{
    font-weight: bolder;
    margin: 5rem auto;
    text-align: center;
}