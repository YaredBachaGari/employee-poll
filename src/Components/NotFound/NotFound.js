import React from "react";
import { Link } from "react-router-dom";
import "./NotFound.css";
import HOC from "../HigherOrderComp/HOC";

const NotFound = ({ AuthUser }) => {
  const { loggedInUser } = AuthUser;
  return (
    <div className="notfound">
      <h1>404 | Page Not Found</h1>
      {loggedInUser?.id ? (
        <Link to="/home" data-testid="gohome">Back to home</Link>
      ) : (
        <Link to="/" data-testid="signIn">Sign In</Link>
      )}
    </div>
  );
};

export default HOC(NotFound);
