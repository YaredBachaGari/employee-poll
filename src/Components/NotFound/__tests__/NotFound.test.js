import React from "react";
import { render, screen } from "@testing-library/react";
import NotFound from "../NotFound";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import { useDispatch } from "react-redux";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
}));

jest.mock("../../../Redux-handler/Actions/Users");
jest.mock("../../../Redux-handler/Actions/Questions");

describe("NotFound Component", () => {
  const mockDispatch = jest.fn();

  beforeEach(() => {
    useDispatch.mockReturnValue(mockDispatch);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });
  it("for non-loggedin users signin button should render and go back btn shoud hide", () => {
    const mockStore = configureStore();
    const store = mockStore({
      AuthUser: {
        loggedInUser: {},
      },
    });
    const { notFoundPage } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NotFound />
        </BrowserRouter>
      </Provider>
    );
   const backbtn = screen.queryByTestId("gohome");
    const signIn = screen.getByTestId("signIn");
    expect(notFoundPage).toMatchSnapshot();
    expect(backbtn).not.toBeInTheDocument();
    expect(signIn).toBeInTheDocument();
  });
  it("for loggedin users signin button should hide and go back btn shoud render", () => {
    const mockStore = configureStore();
    const store = mockStore({
      AuthUser: {
        loggedInUser: {id:1, username:"john"},
      },
    });
    const { notFoundPage } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NotFound AuthUser={store.AuthUser} />
        </BrowserRouter>
      </Provider>
    );
   const backbtn = screen.getByTestId("gohome");
    const signIn = screen.queryByTestId("signIn");
    expect(notFoundPage).toMatchSnapshot();
    expect(backbtn).toBeInTheDocument();
    expect(signIn).not.toBeInTheDocument();
  });
});
