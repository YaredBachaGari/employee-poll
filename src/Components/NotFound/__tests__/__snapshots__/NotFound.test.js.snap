// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component for loggedin users signin button should hide and go back btn shoud render 1`] = `undefined`;

exports[`NotFound Component for non-loggedin users signin button should render and go back btn shoud hide 1`] = `undefined`;
